readthedocs/organizations/models.py

```diff
@@ -26,6 +29,21 @@
 log = structlog.get_logger(__name__)
 
 
+def _upload_organization_avatar_to(instance, filename):
+    extension = filename.split(".")[-1].lower()
+    return f"avatars/organizations/{instance.pk}.{extension}"
```

Comment:
We use the `slug` of the object in other buckets to generate the path. It will be easier to find the file when debugging as well.

```suggestion
    return f"avatars/organizations/{instance.slug}.{extension}"
```

Reply:
We don't change slug via support on purpose. We always tell people to re-create the object. The project slug in .com is the combination of "organization slug + project slug"

Reply:
> So, you change the slug from organization-a to somethingelse and then all the URLs for the projects keep using the old slug? Like organization-a-project-a.readthedocs-hosted.com but the organization slug is now somethingelse?

yep, nothing breaks, and new projects derivate from the new slug.

> We don't want that and that's why we said we won't allow changing organization slugs, even on support requests.

But we do change on support request, and has been done several times for valid use cases.

Reply:
I mentioned enumeration in the description. But I do feel like is very low risk, images are not sensitive information, we are already leaking the name of the organization in other parts of the code when a version is public for example.

Reply:
True. I guess I lean towards slugs still though, it's a middle ground and hashed filenames seems unnecessary.

Images are maybe sensitive if the org doesn't have public anything. And enumeration is just a public list of our users. Slugs skip both these concerns.

And moreso, the downside is really only to organizations changing slugs, which already has a lot of caveats and is an infrequent edge case. I'm fine absorbing this to avoid being, or appearing at least, leaky.

Reply:
I'm :+1: on not using pks, that would make it trivial to see every user that we have, which is definitely bad.

Reply:
If we don't want to use pks, I'd suggest using an UID instead of the slug, having a slug introduces edge cases when an organization slug is renamed.

Reply:
> If it feels like much more work, the rest of us are all okay with just using the org slug.

👍🏼

Reply:
Yeah UUID does avoid all that.

But we're all saying the slug is fine because the risk here doesn't seem significant. The cost of this discussion and spending more time engineering a solution for an edge case will overshadow the cost of spending a few extra minutes each year telling a couple users they need to update their avatar.

Reply:
The cost of having to deal the avatars randomly changing when there is a slug change is greater than implementing a good solution from the start that only requires a couple of lines of code. And as mentioned, this edge case allows for random users changing the avatar from another organization, so the problems caused are not insignificant.